variant_remapping_tools/reads_to_remapped_variants.py

```diff
@@ -64,32 +63,55 @@ def is_read_valid(read, counter, flank_length, score_cutoff, diff_cutoff):
     return True
 
 
-def process_bam_file(bam_file_path, output_file, old_ref_allele_output_file, flank_length, score_perc, diff_AS_XS_perc):
+def fetch_bases(fasta, contig, length, start):
+    zero_base_start = start - 1
+    end = zero_base_start + length
+    new_ref = pysam.FastaFile(fasta).fetch(reference=contig, start=zero_base_start, end=end)
```

Comment:
Initialising a `FastaFile` instance every time has got to be very expensive. Could this be rearranged to only initialise this once and pass a ready instance into this method?

(In fact, once this is done, this method might not be necessary anymore, as fetching a subsection would just be a one liner.)

Reply:
I would prefer keeping the `fetch_bases` because of the tests (make sure we are getting the desired bases) and also to handle the 1-base to 0-base there. Let me know what you think.

Reply:
Yep, absolutely—now that `FastaFile` is initialised outside, I take absolutely no issue with keeping `fetch_bases` as a separate function.